components/ui/tabcounter/src/main/java/mozilla/components/ui/tabcounter/TabCounterMenu.kt

```diff
@@ -0,0 +1,150 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.ui.tabcounter
+
+import android.content.Context
+import androidx.core.content.ContextCompat.getColor
+import mozilla.components.browser.menu2.BrowserMenuController
+import mozilla.components.concept.menu.MenuController
+import mozilla.components.concept.menu.candidate.DrawableMenuIcon
+import mozilla.components.concept.menu.candidate.MenuCandidate
+import mozilla.components.concept.menu.candidate.TextMenuCandidate
+import mozilla.components.concept.menu.candidate.TextStyle
+
+open class TabCounterMenu(
+    context: Context,
+    private val onItemTapped: (Item) -> Unit
+) {
+    open class Item {
+        object CloseTab : Item()
+        object NewTab : Item()
+        object NewPrivateTab : Item()
+    }
+
+    companion object {
+        fun create(context: Context): TabCounterMenu {
+            return TabCounterMenu(
+                context = context,
+                onItemTapped = { item ->
+                    when (item) {
+                        is Item.CloseTab -> {
+                            // replace alllll this:
```

Comment:
yeah, this lambda should live in the application which will create the `TabCounterMenu` and specify/pass `onItemTapped`.